src/dotnet/OpenAPI.Validator/Validation/LongRunningOperationsOptionsValidator.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using OpenAPI.Validator.Model;
+using OpenAPI.Validator.Properties;
+using OpenAPI.Validator.Validation.Core;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenAPI.Validator.Validation
+{
+    public class LongRunningOperationsOptionsValidator : TypedRule<Dictionary<string, Operation>>
+    {
+        /// <summary>
+        /// Id of the Rule.
+        /// </summary>
+        public override string Id => "R2010";
+
+        /// <summary>
+        /// Violation category of the Rule.
+        /// </summary>
+        public override ValidationCategory ValidationCategory => ValidationCategory.SDKViolation;
+
+        /// <summary>
+        /// The template message for this Rule. 
+        /// </summary>
+        /// <remarks>
+        /// This may contain placeholders '{0}' for parameterized messages.
+        /// </remarks>
+        public override string MessageTemplate => Resources.LongRunningOperationsOptionsValidatorMessage;
+
+        /// <summary>
+        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// </summary>
+        public override Category Severity => Category.Warning;
+
+        /// <summary>
+        /// What kind of open api document type this rule should be applied to
+        /// </summary>
+        public override ServiceDefinitionDocumentType ServiceDefinitionDocumentType => ServiceDefinitionDocumentType.ARM;
+
+        /// <summary>
+        /// What kind of change implementing this rule can cause.
+        /// </summary>
+        public override ValidationChangesImpact ValidationChangesImpact => ValidationChangesImpact.SDKImpactingChanges;
+
+        /// <summary>
+        /// The rule could be violated by a model referenced by many jsons belonging to the same
+        /// composed state, to reduce duplicate messages, run validation rule in composed state
+        /// </summary>
+        public override ServiceDefinitionDocumentState ValidationRuleMergeState => ServiceDefinitionDocumentState.Individual;
+
+        /// <summary>
+        /// Validates if a Post LRO Operation with return value has long running operation options extensions enabled.
+        /// </summary>
+        /// <param name="operation"></param>
+        /// <param name="context"></param>
+        /// <returns>true if a Post LRO Operation with return value has long running operation options extensions enabled. false otherwise.</returns>
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Operation> operationPaths, RuleContext context)
+        {
+            Operation operation = null;
+            // Check if there is a post operation. We are not considering any other operation for this rule.
+            if (operationPaths.TryGetValue("post", out operation))
```

Comment:
what comes in operationsPaths? could there be multiple POST operations?

Reply:
No. It cannot have more than one post. Autorest will crash